testflinger_agent/job.py

```diff
@@ -78,24 +79,40 @@ def run_test_phase(self, phase, rundir):
         except Exception as e:
             logger.exception(e)
         finally:
-            with open(os.path.join(rundir, "testflinger-outcome.json")) as f:
-                outcome_data = json.load(f)
+            self.update_phase_results(
+                results_file, phase, exitcode, output_log, serial_log
+            )
+        sys.exit(exitcode)
+
+    def update_phase_results(
```

Comment:
Looks like this should be private?

Reply:
Admittedly, I'm often a little apathetic about "private" methods when it comes to python since it's a lie. But sure, easy enough. :)
I pushed a change to put a "_" before it.